packages/rs-sdk/src/platform/fetch_many.rs

```diff
@@ -141,24 +145,43 @@ where
         sdk: &Sdk,
         query: Q,
     ) -> Result<O, Error> {
-        let request = query.query(sdk.prove())?;
+        let request = &query.query(sdk.prove())?;
+        let closure = |settings: RequestSettings| async move {
+            let grpc_response = request
+                .clone()
+                .execute(sdk, settings)
+                .await
+                .map_err(|e| e.inner_into())?;
 
-        let response = request
-            .clone()
-            .execute(sdk, RequestSettings::default())
-            .await // TODO: We need better way to handle execution response and errors
-            .map(|execution_response| execution_response.into_inner())
-            .map_err(|execution_error| execution_error.into_inner())?;
+            let ExecutionResponse {
```

Comment:
just put it directly to `request.clone().execute(sdk, settings)`

Reply:
Done, also improved logs